File: php/src/ItemUpdater/NormalItemUpdater.php
<?php

declare(strict_types=1);

namespace GildedRose\ItemUpdater;

use GildedRose\Item;
use GildedRose\ItemUpdater\Abstract\AbstractItemUpdater;
use GildedRose\ItemUpdater\Interface\ItemUpdaterInterface;

class NormalItemUpdater extends AbstractItemUpdater implements ItemUpdaterInterface
{
    /**
     * @param Item $item
     *
     * @return void
     */
    public function updateQuality(Item $item): void
    {
        $this->decreaseQuality($item, $this->getQualityDecrease($item));
    }
}

File: Java/src/main/java/com/gildedrose/NormalItemUpdater.java
package com.gildedrose;

public class NormalItemUpdater extends BaseItemUpdater {
    @Override
    public void update(Item item) {
        decreaseSellIn(item);
        if (item.sellIn <0) {
            decreaseQuality(item, 2);
        }
        else {
            decreaseQuality(item);
        }
    }
}
